lib/chart/widgets/chart_widget.dart

```diff
@@ -16,40 +16,64 @@ class _ChartWidget<T> extends StatelessWidget {
   final double? width;
   final ChartState<T?> state;
 
+  EdgeInsets get _horizontalItemPadding =>
+      state.itemOptions.padding.copyWith(top: 0.0, bottom: 0.0);
+
+  double? get _minBarWidth => state.itemOptions.minBarWidth;
+
+  double? get _maxBarWidth => state.itemOptions.maxBarWidth;
+
+  double _wantedItemWidthNonScrollable() {
+    return max(_minBarWidth ?? 0.0, _maxBarWidth ?? 0.0);
+  }
+
+  double _wantedItemWidthForScrollable(double frameWidth) {
+    final visibleItems = state.behaviour.visibleItems;
+    if (visibleItems == null) {
+      return _wantedItemWidthNonScrollable();
+    }
+
+    return state.itemOptions
+        .widthCalculator(visibleItems, _horizontalItemPadding, frameWidth);
+
+    // if (_maxBarWidth == null) {
+    //   return max(_minBarWidth ?? 0.0, calculatedItemWidth);
+    // } else {
+    //   return min(_maxBarWidth!, max(_minBarWidth ?? 0.0, calculatedItemWidth));
+    // }
+  }
+
   @override
   Widget build(BuildContext context) {
     return LayoutBuilder(
       builder: (context, constraints) {
-        // What size does the item want to be?
-        final _wantedItemWidth = state.data.items.foldIndexed<double>(0.0,
-            (index, double prevValue, _) {
-          return max(
-              prevValue,
-              max(state.itemOptions.minBarWidth ?? 0.0,
-                  state.itemOptions.maxBarWidth ?? 0.0));
-        });
-
-        final _width =
+        final frameWidth =
             constraints.maxWidth.isFinite ? constraints.maxWidth : width!;
-        final _height =
+        final frameHeight =
             constraints.maxHeight.isFinite ? constraints.maxHeight : height!;
 
-        final _listSize = state.data.listSize;
+        // Used for smooth transition between scrollable and non-scrollable chart
+        final sizeTween = Tween(
+          begin: _wantedItemWidthNonScrollable(),
+          end: _wantedItemWidthForScrollable(frameWidth),
```

Comment:
You should also pass `state.defaultPadding` to this, or just suptract horizontal paddings. Since not all of the width is for items. 

This is why I can see a bit of 8th item here when I turn on Axis values:
![simulator_screenshot_19DBB775-2987-42BF-9E26-AD2661FE8386](https://user-images.githubusercontent.com/59824861/212716712-6a048087-92e5-4559-9fda-5765474c8d37.png)

Reply:
Ok, I didn't notice `defaultPadding` before, but before it wasn't used when the width of the chart was calculated either. I've added that as well but if you could check that I done it right